packages/client/screens/AddContactScreen.tsx

```diff
@@ -0,0 +1,258 @@
+import { ContactsGateway } from "../gateways/contacts_gateway";
+import { AuthContext } from "../components/contexts/AuthContext";
+
+export default function AddContactScreen(props: SettingsStackScreenProps<"AddContactScreen">) {
+    const contactsGateway = new ContactsGateway();
+
+    const { token, user } = React.useContext(AuthContext);
+
+    type FormValuesType = {
+        contactName: string;
+        ethWallets: string[];
+        btcWallets: string[];
+    };
+
+    const initialValues: FormValuesType = {
+        contactName: "",
+        ethWallets: [],
+        btcWallets: [],
+    };
+
+    type addWalletFieldArrayProps = {
+        values: FormValuesType;
+        handleChange: any;
+        currencyType: CurrencyType;
+        errors: FormikErrors<FormValuesType>;
+        touched: FormikTouched<FormValuesType>;
+    };
+
+    function AddWalletFieldArray({ values, handleChange, currencyType, errors, touched }: addWalletFieldArrayProps) {
+        const wallets = currencyType === CurrencyType.BITCOIN ? values.btcWallets : values.ethWallets;
+        const walletListString = currencyType === CurrencyType.BITCOIN ? "btcWallets" : "ethWallets";
+        const currencyIcon = currencyType === CurrencyType.BITCOIN ? faBitcoin : faEthereum;
+        const iconColor = currencyType === CurrencyType.BITCOIN ? "#F7931A" : "#3C3C3D";
+        const currencyErrors = currencyType === CurrencyType.BITCOIN ? errors.btcWallets : errors.ethWallets;
+        const currencyTouched = currencyType === CurrencyType.BITCOIN ? touched.btcWallets : touched.ethWallets;
+
+        const [isCollapsed, setIsCollapsed] = React.useState<boolean>(true);
+
+        return (
+            <>
+                <Pressable
+                    onPress={() => {
+                        setIsCollapsed(!isCollapsed);
+                    }}
+                >
+                    <HStack style={{ marginTop: 40 }}>
+                        <FontAwesomeIcon style={{ marginRight: 10 }} color={iconColor} icon={currencyIcon} size={26} />
+                        <Text fontWeight={"semibold"} size={"title3"}>
+                            {titleCase(currencyType)} Wallets
+                        </Text>
+                        {isCollapsed ? (
+                            <FontAwesomeIcon style={styles.chevronIcon} size={18} icon={farChevronUp} />
+                        ) : (
+                            <FontAwesomeIcon style={styles.chevronIcon} icon={farChevronDown} size={18} />
+                        )}
+                    </HStack>
+                </Pressable>
+                {/* @ts-expect-error this is a known type error in the dependency a pr was made to address the issue but never merged */}
+                <Collapsible collapsed={isCollapsed}>
+                    <FieldArray
+                        name={walletListString}
+                        render={(arrayHelpers) => (
```

Comment:
Maybe what the list item actually is like wallets is a better name?

```suggestion
                        render={(wallets) => (
```